addons/mail/models/res_users.py

```diff
@@ -278,14 +279,19 @@ def systray_get_activities(self):
         for activity in activities:
             record = self.env[activity.res_model].browse(activity.res_id)
             activities_by_record_by_model_name[activity.res_model][record] += activity
-        model_ids = list({self.env["ir.model"]._get(name).id for name in activities_by_record_by_model_name.keys()})
-        user_activities = {}
+        activities_by_model_name = defaultdict(lambda: self.env["mail.activity"])
```

Comment:
Not related to this task (more to a fix in 16 for people trying to fetch 100K activities): we could add a limit to the initial search, like fetching 1000 activities maximum. Anyway if you have more of them, not sure it still makes sense to display them, so would say we could add a commit adding a limit for performance (and readability) reason.

Reply:
Skipped for now as it may be better to do it in 16 as you have suggested. And also because I have a question, do we hard code that value or we make it configurable ?

Reply:
Not sure people would want to configure it, but indeed we could have a configuration parameter like
```
    search_limit = int(self.env['ir.config_parameter'].sudo().get_param('mail.activity.systray.limit', 1000))
```

If limit is 0, should be managed like None

Reply:
I would do it in 17, and maybe backport it afterwards. That way we have working proposal for 17.